specs/addresses/cross-chain-interoperable-addresses-spec.md

```diff
@@ -1,197 +1,449 @@
 
-Human-readable name
-: `1::eip155:1:0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045`
+### Human-readable name resolution
+The CAIP-2 namespace is to be rendered alongside the CAIP-10 formatted address and the checksum:
 
-### `2` : ERC-3770 chain name resolution only
-The human-readable name in this approach is consistent with the address format defined in ERC-3770, and similarly depends on the EF's mapping of chain ids to names maintained in github at: https://github.com/ethereum-lists/chains
+```
+<CAIP-10 formatted address>@<CAIP-2 namespace>#<checksum>
+```
 
-#### Human-readable name format
+### Examples
 
-```bnf
-<human readable name> ::= 2::<short chain name>:<EIP-55 formatted address>
+Human-readable name:
+`0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045#6164DA10@eip155:1#618ad0d1`:
+
+First word:
+```
+MSB                                                            LSB
+0x80006164DA100000000000000000000000000000000000000000000000000000
+  ^^^^------------------------ 255-240 Interoperable Address version
+      ^^^^^^^^ --------------- 239-208 Checksum
+              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+              \--------------- 207-0   Not used
 ```
-Since wallet software can always validate the checksum, it MUST be removed.
 
-EIP-55 or applicable scheme is used for canonicalization only.
+Second word (second field):
+```
+MSB                                                            LSB
+0x0100000000000000000000000000000000000000000000000000000000000002
+  ^^----------------------------------------------------------------- 255-249 bytes array payload
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ---- 9-249   padding
+                                                                ^^ -- 0-8    2*1 (payload length)
+```
 
-#### Examples:
+Third word (third field):
+```
+MSB                                                            LSB
+0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045000000000000000000000028
+  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  ------------------------- 255-97 bytes array payload
+                                          ^^^^^^^^^^^^^^^^^^^^^^ ---- 9-96   padding
+                                                                ^^ -- 0-8    2*20 (payload length)
+```
 
-Machine address
-: `2:::eip155:1:0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045#bc797def`
+TODO: example with multi-word fields
 
-Human-readable name
-: `2::eth:0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045`
+## `0xC000`: Interoperable Canonical Format With Resolver Info: arbitrary length addresses, arbitrary length name-resolving contract specification
```

Comment:
I'm not sure I see the usefulness of this - it replaces the non-human-readable address with a human-readable name, but then adds a non-human-readable address that specifies how the name is resolved. The net result is an address that's longer and no easier to read/verify.